### ISSUE TITLE
Two whitespaces on redirect_uri works as a wildcard

### ISSUE THREAD

**Issue by Maks3w** [bshaffer/oauth2-server-php#663]
Given this redirect_uri `" http://localhost"` makes valid any `$supplied_redirect_uri` if `require_exact_redirect_uri = false`.

Because after explode/split the string by the whitespace character
https://github.com/bshaffer/oauth2-server-php/blob/40e2e32786a76463b721e84d1ee7e7959abcd7c0/src/OAuth2/Controller/AuthorizeController.php#L336

```php
$registered_uri = "";
$inputUri = "any";
var_dump(strcasecmp(substr($inputUri, 0, strlen($registered_uri)), $registered_uri) === 0); // true
```
https://github.com/bshaffer/oauth2-server-php/blob/40e2e32786a76463b721e84d1ee7e7959abcd7c0/src/OAuth2/Controller/AuthorizeController.php#L346

Is this behavior expected or should be protected against empty `$registered_uri` values?

I found this behavior can be achieved accidentally if `$redirect_uri` is constructed by implode a `null` value (implode(" ", [null, "http://localhost"]);

**Comment by bshaffer** [comment]
No, this is not intended behavior, since the whole point in registering a redirect URI (or set of them) is to prevent a wildcard URI. Otherwise, you'd just leave it blank.

We should add an `array_filter` call after `explode`, to remove any empty values and prevent this from happening.

**Comment by Maks3w** [comment]
Proposed #665 as fix.

**Comment by bshaffer** [comment]
fixed in #695
